lib/ipfs.c

```diff
@@ -0,0 +1,74 @@
+/***************************************************************************
+ *                                  _   _ ____  _
+ *  Project                     ___| | | |  _ \| |
+ *                             / __| | | | |_) | |
+ *                            | (__| |_| |  _ <| |___
+ *                             \___|\___/|_| \_\_____|
+ *
+ * Copyright (C) 2020 - 2022, Daniel Stenberg, <daniel@haxx.se>, et al.
+ * Copyright (C) 2019, Björn Stenberg, <bjorn@haxx.se>
+ *
+ * This software is licensed as described in the file COPYING, which
+ * you should have received as part of this distribution. The terms
+ * are also available at https://curl.se/docs/copyright.html.
+ *
+ * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+ * copies of the Software, and permit persons to whom the Software is
+ * furnished to do so, under the terms of the COPYING file.
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+ * KIND, either express or implied.
+ *
+ ***************************************************************************/
+
+#include "curl_setup.h"
+
+#ifndef CURL_DISABLE_IPFS
+
+#include "urldata.h"
+#include <curl/curl.h>
+
+/* The last #include file should be: */
+#include "memdebug.h"
+
+/*
+ * Forward declarations.
+ */
+
+static CURLcode ipfs_do(struct Curl_easy *data, bool *done);
+
+/*
+ * MQTT protocol handler.
+ */
+
+const struct Curl_handler Curl_handler_ipfs = {
+  "IPFS",                             /* scheme */
+  ZERO_NULL,                    /* setup_connection */
+  ipfs_do,                            /* do_it */
```

Comment:
The main feedback I need is: If my feature is essentially just rewriting an `ipfs://<CID>/<PATH>` url to an `https://<CID>.ipfs.dweb.link/<PATH>` url, should I have an ipfs_do function that looks like a normal program that happens to use curl https public APIs -- or should I implement a full Curl_handler that passes through to the `http_do` and similar functions for everything after the URL rewrite.